### PR TITLE
[#3715] Add dependency question validation on publishing survey

### PR DESCRIPTION
**Pull request by tangrammer**
#### Screenshots (if appropriate)

Screenshot 2021-01-14 at 18 44 09

Screenshot 2021-01-18 at 12 57 22

#### locally tested following this "trick"
replace this line https://github.com/akvo/akvo-flow/pull/3731/files?file-filters%5B%5D=.handlebars&file-filters%5B%5D=.js&file-filters%5B%5D=.properties#diff-4aa8b0a0ea54a2a4c2515c81c7908acf3c0ae84038c9cbdba793642649da6c72R658 with following code ...
```js
    const questionGroups = FLOW.store.filter(FLOW.QuestionGroup, item => item.get('surveyId') == surveyId).map((item) => FLOW.store.find(FLOW.QuestionGroup, item.get("id")));

    const questionGroupClientId = questionGroups[0]._data.attributes.keyId;
    const questionsSelectedGroup = questions.filter(q => q._data.attributes.questionGroupId === questionGroupClientId);
    const data = {};
    data[questionGroupClientId] = [questionsSelectedGroup[0]._data.attributes.keyId, questionsSelectedGroup[2]._data.attributes.keyId];
return data;

```

then:
1- go to http://localhost:8888/admin/
2 - select: Folder with a few large datasets > BAR-handpump
3 - click publish

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

**Comment by tangrammer** [comment]
@muloem refactor code following your advise!

**Review by tangrammer — COMMENTED** [review]
implementing now the jquery approach

File: Dashboard/app/js/templates/navSurveys/edit-questions.handlebars [hunk 1]
```diff
@@ -35,9 +35,10 @@
     <!-- start list of question groups -->
     {{#each questionGroup in FLOW.questionGroupControl.arrangedContent}}
       {{#view FLOW.QuestionGroupItemView contentBinding="questionGroup"}}
-        <div class="questionGroupBlock">
-          <header><span class="qtnGroupHead">{{t _group}} {{view.content.order}}</span>
-
+            <div class="questionGroupBlock">
+                {{#if view.amQuestionGroupPublishingError}} <div class="errorQuestionGroupTop"> </div>{{/if}}
```
[the remark this answers is not in the document]
Comment (tangrammer) [reply Dashboard/app/js/templates/navSurveys/edit-questions.handlebars]: solved by https://github.com/akvo/akvo-flow/pull/3731/commits/8f23dcc08938b3bf0dd8e08021fd7344575ca7e1

**Comment by tangrammer** [comment]
@muloem  ready to review!

File: Dashboard/app/js/lib/controllers/survey-controllers.js [hunk 2]
```diff
@@ -641,18 +642,44 @@ FLOW.surveyControl = Ember.ArrayController.create(observe({
     this.set('newLocaleSurvey', this.find(item => item.get('keyId') === newLocaleId));
   },
 
+
+  validateSurveyToBePublished(surveyId) {
+
+    const questions = FLOW.store.filter(FLOW.Question).map((item) => FLOW.store.find(FLOW.Question, item.get("id"))).filter(o => o.get('surveyId') == surveyId);
```
[the remark this answers is not in the document]
Comment (tangrammer) [reply Dashboard/app/js/lib/controllers/survey-controllers.js]: Accepted suggestion! 😄

**Review by muloem — APPROVED** [review]

**Pull request merged**